docs/kubernetes/kctlr-use-bigip-k8s.rst

```diff
@@ -0,0 +1,188 @@
+
+Tasks
+-----
+
+Complete the following tasks to add a BIG-IP device to a `Kubernetes`_ `Cluster Network`_ using `flannel`_.
+
+===== ==================================================================================
+Step  Task
+===== ==================================================================================
+1.    :ref:`k8s-vxlan-setup`
+----- ----------------------------------------------------------------------------------
+2.    :ref:`k8s-flannel-setup`.
+----- ----------------------------------------------------------------------------------
+3.    :ref:`k8s-bigip-node`
+----- ----------------------------------------------------------------------------------
+4.    :ref:`k8s-assign-ip`
+===== ==================================================================================
+
+\
+
+.. _k8s-vxlan-setup:
+
+Create a VXLAN tunnel on the BIG-IP device
+------------------------------------------
+
+#. Create a new BIG-IP VXLAN profile.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ create net tunnels vxlan fl-vxlan port 8472 flooding-type none
+
+#. Verify creation of the VXLAN profile.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ list net tunnels vxlan fl-vxlan
+
+
+#. Create a new BIG-IP VXLAN tunnel.
+
+   - Use the BIG-IP VTEP (internal) address as the VXLAN ``local-address``.
+   - Set the ``key`` to ``1`` to grant the BIG-IP device access to all Kubernetes/Flannel projects and subnets.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ create net tunnels tunnel flannel_vxlan key 1 profile fl-vxlan local-address 172.16.1.3
+
+#. Verify creation of the VXLAN tunnel.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ list net tunnels tunnel flannel_vxlan
+
+
+.. _k8s-flannel-setup:
+
+Deploy flannel for Kubernetes
+-----------------------------
+
+Most users deploy flannel as a `network addon`_ when setting up ``kubeadm``. See Kubernetes` :k8sdocs:`Installing a Pod network <setup/independent/create-cluster-kubeadm/#pod-network>` guide for instructions.
+
+**If you want to install flannel manually**, take the steps below.
+
+#. (Optional) Edit the flannel `kube-flannel manifest`_.
+
+   - ``Backend``: Defaults to ``vxlan``; see `Backends <https://github.com/coreos/flannel/blob/master/Documentation/backends.md>`_ for additional information.
+   - ``Network``: Should match the Pod network CIDR. Flannel will populate this information automatically.
+
+#. Use :command:`kubectl apply` to deploy the flannel manifest.
+
+   .. code-block:: console
+
+      kubectl apply -f kube-flannel.yaml --namespace=kube-system
+
+   \
+
+.. seealso::
+
+   For more information about flannel and Kubernetes, see `Using flannel with Kubernetes`_.
+
+#. Verify flannel deployment.
+
+   Check the Node resource(s) for the following Annotations: ::
```

Comment:
Should we include the example kubectl command to guide the user on how to get this?